client-spark/spark3/src/main/java/org/apache/spark/shuffle/reader/RssShuffleReader.java

```diff
@@ -119,6 +120,9 @@ public RssShuffleReader(
     this.partitionToShuffleServers = rssShuffleHandle.getPartitionToServers();
     this.rssConf = rssConf;
     this.dataDistributionType = dataDistributionType;
+    // This mechanism of expectedTaskIdsBitmap filter is to filter out the most of data.
+    // especially for AQE skew optimization
+    this.expectedTaskIdsBitmapFilterEnable = mapEndIndex == Integer.MAX_VALUE ? false : true;
```

Comment:
If this is the last reduce partition, may the range of mapId be [n, Integer.MAX_VALUE]?

Reply:
So do we need to use the `startMapIndex==0` and `mapEndIndex==max_value` to judge?

Reply:
Updated.